enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,564 @@
+
+## Summary
+
+Static IP addresses are emerging as a common requirement in environments where
+the usage of DHCP violates corporate security guidelines.  Additionally, many 
+users which require static IPs also require the use of the IPI installer. 
+The proposal described in this enhacement discusses the implementation of
+of assiging static IPs at both day 0 and day 2.
+
+## Motivation
+
+Users of OpenShift would like the ability to provision vSphere IPI clusters with static IPs.
+
+- https://issues.redhat.com/browse/OCPPLAN-9654
+
+### User Stories
+
+As an OpenShift administrator, I want to provision nodes with static IP addresses so that I can comply with my organization's security requirements.
+
+As an OpenShift administrator, I want to provision static IP addresses with the IPI installer so that I can reduce the complexity of certifying tools required to provision OpenShift.
+
+As an OpenShift administrator, I want to scale nodes with static IPs so that I can meet capacity demands as well as respond to disaster recovery scenarios.
+
+As an OpenShift engineer, I want to know if a IPI installation was performed so that support can determine the IP assignment strategy for a given cluster.
+
+
+### Goals
+
+- All nodes created during the installation are configured with static IPs
+      Rationale: Many environments, due to security policies, do not allow DHCP.
+
+- The IPI installation method is able to provide static IPs to the nodes
+  Rationale: Some users must qualify each tool used in their environment. 
+  Leveraging IPI greatly reduces the number of tools required to provision
+  a cluster.
+- 
+
+### Non-Goals
+
+- OpenShift will not be responsible for managing IP addresses.
+- Multiple interfaces, IP configurations, and/or routes support is out of scope.
+
+## Proposal
+
+### Static IPs Configured at Installation
+
+To faciliate the configuration of static IP address, nmstate definitions are created for each node in the install-config.yaml. This follows a similar pattern set forth in the [Baremetal IPI Network Configuration](https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-ipi-network-configuration.md#user-facing-api) enhancement. 
+
+This enhancement expands on the [`hosts` slice](https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-ipi-network-configuration.md#user-facing-api) by associating hosts with roles and, optionally, a failure domain.
+
+For the bootstrap and control plane nodes, static IPs are passed to the node via the `guestinfo.afterburn.initrd.network-kargs` extraconfig parameter.  [Afterburn](https://github.com/coreos/afterburn/blob/main/src/providers/vmware/amd64.rs) recognizes this parameter when the node initially boots. 
+
+For compute nodes, the installer will produce will produce `Machine` resources in addition to a `MachineSet`. The `MachineSet` will be set to `0` replicas to aid in the day 2 scaling of nodes.  
```

Comment:
The idea here isn't to have the installer create the compute VM instances.  It is creating worker machine resources that are reconciled when the machine API starts.  The primary difference is that when static IP addresses are to be used, the installer creates the machine resources with the IP configuration already configured, rather than machinesets.

Reply:
I added some clarifying language to this section.